datalad/install.py

```diff
@@ -0,0 +1,396 @@
+        )
+        policy = r.stdout
+        devel_annex_version = None
+        current_annex_version = None
+        prev = None
+        for line in policy.splitlines():
+            if "/debian-devel " in line:
+                assert prev is not None
+                if "ndall" in prev:
+                    assert devel_annex_version is None
+                    devel_annex_version = prev.split()[0]
+            if "***" in line:
+                assert current_annex_version is None
+                current_annex_version = line.split()[1]
+            prev = line
+        assert devel_annex_version is not None, "Could not find devel annex version"
+        assert current_annex_version is not None, "Could not find current annex version"
+        if (
+            subprocess.run(
+                [
+                    "dpkg",
+                    "--compare-versions",
+                    devel_annex_version,
+                    "gt",
+                    current_annex_version,
+                ]
+            ).returncode
+            == 0
+        ):
+            subprocess.run(
+                [
+                    "sudo",
+                    "apt-get",
+                    "install",
+                    f"git-annex-standalone={devel_annex_version}",
+                ],
+                check=True,
+            )
+        else:
+            print(
+                f"I: devel version {devel_annex_version} is not newer than installed {current_annex_version}",
+                file=sys.stderr,
+            )
+        self.post_install()
+
+    def install_via_deb_url(self, url):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            debpath = os.path.join(tmpdir, "git-annex.deb")
+            subprocess.run(["wget", "-O", debpath, url], check=True)
+            subprocess.run(["sudo", "dpkg", "-i", debpath], check=True)
+        self.post_install()
+
+    def install_via_autobuild(self):
+        systype = platform.system()
+        if systype == "Linux":
+            self._install_via_autobuild_or_snapshot_linux("autobuild/amd64")
+        elif systype == "Darwin":
+            self._install_via_autobuild_or_snapshot_macos(
+                "autobuild/x86_64-apple-yosemite"
+            )
+        else:
+            raise RuntimeError(f"E: Unsupported OS: {systype}")
+
+    def install_via_snapshot(self):
+        systype = platform.system()
+        if systype == "Linux":
+            self._install_via_autobuild_or_snapshot_linux("linux/current")
+        elif systype == "Darwin":
+            self._install_via_autobuild_or_snapshot_macos("OSX/current/10.10_Yosemite")
+        else:
+            raise RuntimeError(f"E: Unsupported OS: {systype}")
+
+    def _install_via_autobuild_or_snapshot_linux(self, subpath):
+        tmpdir = tempfile.mkdtemp(prefix="ga-")
+        self.annex_bin = os.path.join(tmpdir, "git-annex.linux")
+        print("I: downloading and extracting under {self.annex_bin}", file=sys.stderr)
```

Comment:
Let's use logging, request a logger for `datalad.install` and if there is no handlers in that logger's "root"  (thus not imported from datalad) it would establish the simple default one (probably dumping to stderr).  This way it would stay nicely consistent with datalad logging when used from "inside" of datalad and not as a script, and would still log nicely if used as a script.

Reply:
Done.